.circleci/config.yml

```diff
@@ -58,18 +91,50 @@ jobs:
       - *install_goreleaser
       - run: goreleaser
 
+  release_images:
+    working_directory: /go/src/github.com/fairwindsops/polaris/
+    docker:
+      - image: quay.io/reactiveops/ci-images:v8.0-stretch
+    steps:
+      - checkout
+      - setup_remote_docker
+      - *set_environment_variables
+      - *docker_build
+      - *docker_push
+
 workflows:
   version: 2
-  test-and-release:
+
+  test:
     jobs:
-      - test:
+      - test
+      - build:
+          requires:
+            - test
           filters:
             branches:
-              only: /.*/
+              only: /pull\/[0-9]+/
+      - build_and_push:
+          context: org-global
+          requires:
+            - test
+          filters:
+            branches:
+              ignore: /pull\/[0-9]+/
```

Comment:
Not sure what this is for, but we don't use this pattern AFAIK

Reply:
For external PRs, circle sets the branch name to `pull-###`

We use this pattern in Polaris so that we `docker build` external PRs, but don't push them.

Reply:
I'm still a little confused. Right now I see:

Branches that are not PRs: build and push
Tags: build and push
PR builds: just build, don't push

My normal pattern is to build and push everything. Rok8s scripts will handle the tagging appropriately. 

This allows anyone building a new feature to test it by pulling the docker file from the SHA that built it. 

If you look at your latest commit, you'll see that images got pushed for the SHA, the branch name, and the build_*.